src/experimental/Db.cpp

```diff
@@ -32,3 +32,117 @@ bool DB::init()
     }
     return true;
 }
+
+bool DB::commit(Entry entry)
+{
+    if (exists(entry)) {
+        return update(entry);
+    } else {
+        return insert(entry);
+    }
+}
+
+bool DB::remove(Entry entry)
+{
+    if (!exists(entry)) {
+        DEB("Error: Entry does not exist.");
+        return false;
+    }
+
+    QString statement = "DELETE FROM " + entry.position.first +
+            " WHERE _rowid_=" + QString::number(entry.position.second);
+    QSqlQuery q(statement);
+    //check result.
+    if (q.lastError().type() == QSqlError::NoError)
+    {
+        DEB("Entry " << entry.position.first << entry.position.second << " removed.");
+        return true;
+    } else {
+        DEB("Unable to delete.");
+        DEB("Query: " << statement);
+        DEB("Query Error: " << q.lastError().text());
+        return false;
+    }
+}
+
+bool DB::exists(Entry entry)
+{
+    if (entry.position.second == 0)
```

Comment:
If you dont like this we can make position a custom type with better names for attributes.
```c++    
struct DP : DataPosition {
    TableName table;
    RowId row_id;
    DP(TableName t, RowId i) : DataPosition(t, i) {}
};
```
now we can write `.table` or `.row_id` for extra clarity for the future